### PR TITLE
Smoke test suite

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
Smoke tests for each test suite were added.

### CODE REVIEW

File: itests/utilities/reward/reward.go [hunk 1]
```diff
@@ -83,3 +83,43 @@ func GetBlockRewardDistribution(suite *f.BaseSuite, addresses testdata.Addresses
 		diffDaoBalance.BalanceInWavesGo, diffDaoBalance.BalanceInWavesScala, diffXtnBalance.BalanceInWavesGo,
 		diffXtnBalance.BalanceInWavesScala), term
 }
+
+type TestData func(suite *f.BaseSuite, addresses testdata.AddressesForDistribution,
```
Comment (nickeskov) [inline itests/utilities/reward/reward.go:87]: Strange naming for the closure or function type. Is it testdata generator function type?

**Review by nickeskov — CHANGES_REQUESTED** [review]

File: itests/alias_tx_api_internal_test.go [hunk 2]
```diff
@@ -153,32 +133,32 @@ func (suite *AliasTxApiSuite) Test_SameAliasDiffAddressesApiNegative() {
 		caseName := utl.GetTestcaseNameWithVersion(name, v)
 		suite.Run(caseName, func() {
 			//send alias tx from account that is in first element of testdata slice
-			tx, _, actualDiffBalanceInWaves := alias_utilities.BroadcastAliasTxAndGetWavesBalances(&suite.BaseSuite,
-				tdSlice[0], v, waitForTx)
-			errMsg := caseName + "Broadcast Alias Tx: " + tx.TxID.String()
+			tx1, _, actualDiffBalanceInWaves1 := alias.BroadcastAliasTxAndGetWavesBalances(&suite.BaseSuite,
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/alias_tx_api_internal_test.go:136]: Such names were used for transactions under test which were sent several times. I think that it's ok for that.

File: itests/utilities/common.go [hunk 3]
```diff
@@ -634,13 +654,25 @@ func SendAndWaitTransaction(suite *f.BaseSuite, tx proto.Transaction, scheme pro
 	connections, err := net.NewNodeConnections(suite.Ports)
 	suite.Require().NoError(err, "failed to create new node connections")
 	defer connections.Close(suite.T())
+
 	connections.SendToNodes(suite.T(), txMsg, scala)
+	suite.T().Log("Tx msg was successfully send to nodes")
 
+	suite.T().Log("Waiting for Tx appears in Blockchain")
 	errGo, errScala := suite.Clients.WaitForTransaction(id, timeout)
+	txInfoRawGo, respGo, _ := suite.Clients.GoClients.HttpClient.TransactionInfoRaw(id)
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/common.go:663]: Yes, it is.

File: itests/utilities/common.go [hunk 4]
```diff
@@ -650,7 +682,18 @@ func BroadcastAndWaitTransaction(suite *f.BaseSuite, tx proto.Transaction, schem
...
 	}
+	suite.T().Log("Tx msg was successfully Broadcast to nodes")
+
+	suite.T().Log("Waiting for Tx appears in Blockchain")
 	errWtGo, errWtScala := suite.Clients.WaitForTransaction(id, timeout)
+	txInfoRawGo, responseGo, _ := suite.Clients.GoClients.HttpClient.TransactionInfoRaw(id)
+	suite.T().Logf("Tx Info Go after waiting: %s, Response Go: %s",
+		GetTransactionJsonOrErrMsg(txInfoRawGo), responseGo.Status)
+	suite.T().Log(errors.Errorf("Errors after waiting: %s", errWtGo))
+	txInfoRawScala, responseScala, _ := suite.Clients.ScalaClients.HttpClient.TransactionInfoRaw(id)
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/common.go:693]: Yes, it is, it is used for logging.

File: itests/utilities/common.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/common.go:663]: But it is also used for debugging.

File: itests/utilities/reward/reward.go (hunk 1 above)
Comment (irina-pereiaslavskaia) [reply itests/utilities/reward/reward.go:87]: This function type is used for common representation of function for getting test data. Renamed to GetTestData.

File: itests/utilities/transfer/transfer.go [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+}
+
+type MakeTx[T any] func(suite *f.BaseSuite, testdata testdata.TransferTestData[T], version byte,
+	waitForTx bool) utl.ConsideredTransaction
+
+func MakeTxAndGetDiffBalances[T any](suite *f.BaseSuite, testdata testdata.TransferTestData[T],
+	version byte, waitForTx bool, makeTx MakeTx[T]) (utl.ConsideredTransaction,
+	utl.AccountsDiffBalancesTxWithSponsorship) {
+	var assetDetails *client.AssetsDetail
+	if testdata.FeeAsset.ToDigest() != nil {
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/transfer/transfer.go:58]: Fix.

File: itests/reward_distribution_rollback_internal_test.go
[the remark this answers is not in the document]
Comment (nickeskov) [reply itests/reward_distribution_rollback_internal_test.go:1]: Got it

File: itests/utilities/transfer/transfer.go [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+	accNumber, _ := utl.AddNewAccount(suite, scheme)
+	tx := TransferringFunds(suite, version, scheme, from, accNumber, amount)
+	require.NoError(suite.T(), tx.WtErr.ErrWtGo, "Reached deadline of Transfer tx in Go")
+	require.NoError(suite.T(), tx.WtErr.ErrWtScala, "Reached deadline of Transfer tx in Scala")
+	// Waiting for changing waves balance.
+	err := node_client.Retry(utl.DefaultTimeInterval, func() error {
+		var balanceErr error
+		balanceGo, balanceScala := utl.GetAvailableBalanceInWaves(suite, utl.GetAccount(suite, accNumber).Address)
+		if balanceScala == 0 && balanceGo == 0 {
+			balanceErr = errors.New("account Waves balance is empty")
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/transfer/transfer.go:161]: Fix.

File: itests/smoke_tx_api_positive_internal_test.go [hunk 7]
```diff
@@ -0,0 +1,322 @@
...
+
+func (suite *SmokeTxApiPositiveSuite) Test_AliasTxApiSmokePositive() {
+	v := byte(testdata.AliasMaxVersion)
+	tdmatrix := testdata.GetAliasPositiveDataMatrix(&suite.BaseSuite)
+	for name, td := range tdmatrix {
+		caseName := utl.GetTestcaseNameWithVersion(name, v)
+		suite.Run(caseName, func() {
+			tx, _, actualDiffBalanceInWaves := alias.BroadcastAliasTxAndGetWavesBalances(&suite.BaseSuite,
+				td, v, true)
+			errMsg := caseName + "Broadcast Alias Tx: " + tx.TxID.String()
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/smoke_tx_api_positive_internal_test.go:35]: Fix in all places.

File: itests/sponsorship_tx_api_internal_test.go [hunk 8]
```diff
@@ -19,122 +20,91 @@ type SponsorshipTxApiSuite struct {
...
+		itx := issue.BroadcastWithTestData(&suite.BaseSuite, reissuable, v, true)
 		tdmatrix := testdata.GetSponsorshipPositiveDataMatrix(&suite.BaseSuite, itx.TxID)
 		for name, td := range tdmatrix {
 			caseName := utl.GetTestcaseNameWithVersion(name, v)
 			suite.Run(caseName, func() {
-				tx, actualDiffBalanceInWaves, actualDiffBalanceInAsset := sponsor_utilities.BroadcastSponsorshipTxAndGetBalances(
-					&suite.BaseSuite, td, v, waitForTx)
+				tx, actualDiffBalanceInWaves, actualDiffBalanceInAsset :=
+					sponsorship.BroadcastSponsorshipTxAndGetBalances(&suite.BaseSuite, td, v, true)
 				errMsg := caseName + "Broadcast Sponsorship tx: " + tx.TxID.String()
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/sponsorship_tx_api_internal_test.go:33]: Fix

File: itests/utilities/issue/issue.go [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+func getAssetsMatrix(suite *f.BaseSuite, data testdata.IssueTestData[testdata.ExpectedValuesPositive],
+	issueVersions []byte, casesCount int) [][]crypto.Digest {
+	waitForTx := false
+	matrix := make([][]crypto.Digest, len(issueVersions))
+	for i := 0; i < len(issueVersions); i++ {
+		matrix[i] = make([]crypto.Digest, casesCount)
+		for j := 0; j < casesCount; j++ {
+			if i == len(issueVersions)-1 && j == casesCount-1 {
+				waitForTx = true
+			}
```
[the remark this answers is not in the document]
Comment (irina-pereiaslavskaia) [reply itests/utilities/issue/issue.go]: Fix.

**Review by nickeskov — APPROVED** [review]

**Pull request merged**